src/core/models/outputs/outputs.c

```
@@ -3,80 +3,313 @@
+        break;
+      case OUTPUT_ALIAS:
+        offset += output_alias_serialize(elm->output->output, offset, output_alias_serialize_len(elm->output->output));
+        break;
+      case OUTPUT_FOUNDRY:
+        offset +=
+            output_foundry_serialize(elm->output->output, offset, output_foundry_serialize_len(elm->output->output));
+        break;
+      case OUTPUT_NFT:
+        offset += output_nft_serialize(elm->output->output, offset, output_nft_serialize_len(elm->output->output));
+        break;
+    }
+  }
+
+  return expected_bytes;
+}
+
+utxo_outputs_list_t *utxo_outputs_deserialize(byte_t buf[], size_t buf_len) {
+  if (buf == NULL || buf_len < 2) {
+    printf("[%s:%d] invalid paramters\n", __func__, __LINE__);
+    return NULL;
+  }
+
+  utxo_outputs_list_t *outputs = utxo_outputs_new();
+
+  size_t offset = 0;
+
+  // number of output entries
+  uint16_t output_cnt = 0;
+  memcpy(&output_cnt, &buf[offset], sizeof(uint16_t));
+  offset += sizeof(uint16_t);
+
+  for (uint8_t i = 0; i < output_cnt; i++) {
+    // create a new output list object
+    utxo_outputs_list_t *new_output = malloc(sizeof(utxo_outputs_list_t));
+    if (!new_output) {
+      printf("[%s:%d] OOM\n", __func__, __LINE__);
+      utxo_outputs_free(outputs);
+      return NULL;
+    }
+    new_output->output = malloc(sizeof(utxo_output_t));
+    if (!new_output->output) {
+      printf("[%s:%d] OOM\n", __func__, __LINE__);
+      free(new_output);
+      utxo_outputs_free(outputs);
+      return NULL;
+    }
+    LL_APPEND(outputs, new_output);
+
+    // get type of output
+    if (buf_len < offset + sizeof(uint8_t)) {
+      printf("[%s:%d] invalid data length\n", __func__, __LINE__);
+      utxo_outputs_free(outputs);
+      return NULL;
+    }
+    new_output->output->output_type = buf[offset];
+
+    // deserialize output
+    switch (new_output->output->output_type) {
```

should handle deprecated and not supported output types as well.

---

So `utxo_outputs_deserialize` function should skip deprecated and unsupported output types or is better to raise an error?
And `utxo_outputs_add` should prevent to add any deprecated or unsupported output type?

---

With unsupported and deprecated types, the deserialized data would be invalid, the deserialize function should throw an error.

> And should `utxo_outputs_add` function prevent to add any deprecated or unsupported output type?

We can have it for better UX and debugging, maybe it's a typo.

---

Done